haddock-api/src/Haddock/GhcUtils.hs

```diff
@@ -63,6 +66,13 @@ getMainDeclBinder (ForD _ (ForeignImport _ name _ _)) = [unLoc name]
 getMainDeclBinder (ForD _ (ForeignExport _ _ _ _)) = []
 getMainDeclBinder _ = []
 
+uniquifyClassSig :: (NamedThing name, SetName name) => Bool -> name -> name
+uniquifyClassSig False = id
+uniquifyClassSig _     = liftA2 setName (updateName . getName) id
+  where
+    updateName = liftA2 setNameUnique id $
+        liftA2 deriveUnique id ((+1) . getKey) . nameUnique
```

Comment:
I see several issues with this function but it doesn't appear to be used at all?!